### PR TITLE
primesieve: add version 11.0

### PR DESCRIPTION
**Pull request by Alex-PLACET**
Specify library name and version:  **primesieve/8.0**

This library provides a state of the art prime numbers generator.
I'm not the author of this library.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Linked issue: https://github.com/conan-io/conan-center-index/issues/14529

### CODE REVIEW

**Comment by kimwalisch** [comment]
@Alex-PLACET Thanks for packaging primesieve, I am the author of that project. I am about to release a new version of primesieve so I googled to see if I could find any issues that users have run into.

I read the code of your pull request and it contains a test (`recipes/primesieve/all/test_package/test_package.cpp`) from primesieve's test suite. However, this test requires the latest code from primesieve's master branch (or the upcoming primesieve-11.0) instead of primesieve-8.0 you have currently packaged. You could use https://github.com/kimwalisch/primesieve/blob/v8.0/test/generate_n_primes1.cpp to fix the issue or even better just package primesieve-11.0 (which will be released this weekend) instead of primesieve-8.0. I think you would only need to increase the version number from 8.0 to 11.0 in your merge request (and update the checksums).

Did you encounter any other issues while packaging primesieve? If so I could fix them before releasing primesieve-11.0.

**Comment by Alex-PLACET** [comment]
@kimwalisch Hi,
Thank you for your comment. I will update the recipe to use the latest version of the library.
You right, I made a mistake by using a test file from the master.
And no, no issue, it work like a charm 👍.
Thank you for your work

File: recipes/primesieve/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+class PackageConan(ConanFile):
+    name = "primesieve"
+    description = "🚀 Fast prime number generator"
```
Comment (uilianries) [inline recipes/primesieve/all/conanfile.py:13]:
```suggestion
    description = "Fast prime number generator"
```

File: recipes/primesieve/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+sources:
+  # Newer versions at the top
```
Comment (uilianries) [inline recipes/primesieve/all/conandata.yml:2]:
```suggestion
```

**Review by uilianries — COMMENTED** [review]
Nice addition, please, remove the emoji from the description as not all terminals could parse it.

**Pull request closed without merging**